From d065e7cb1cd6204fad1f1e9f6aa638dae79748d7 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 5 Sep 2018 17:30:59 +0200
Subject: [PATCH] PONRTSYS-1977: gpio: intel-sso: Add GPIO get function

This adds a simple GPIO get function which will be used by the Linux GPIO system.
The hardware does not allow to configure the GPIOs as input, but it already
helps to read out the value to which the output GPIO is configured.
---
 drivers/gpio/gpio-intel-sso.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpio/gpio-intel-sso.c b/drivers/gpio/gpio-intel-sso.c
index b29aec2d536b..9b3c19b7898b 100644
--- a/drivers/gpio/gpio-intel-sso.c
+++ b/drivers/gpio/gpio-intel-sso.c
@@ -121,6 +121,15 @@ sso_gpio_dir_out(struct gpio_chip *chip, unsigned int offset, int value)
 	return sso_gpio_update_bit(priv->mmap, SSO_CPU, offset, value);
 }
 
+static int sso_gpio_get(struct gpio_chip *chip, unsigned int offset)
+{
+	struct sso_gpio_priv *priv = gpiochip_get_data(chip);
+	u32 reg_val;
+
+	regmap_read(priv->mmap, SSO_CPU, &reg_val);
+	return !!(reg_val & BIT(offset));
+}
+
 static void sso_gpio_set(struct gpio_chip *chip, unsigned int offset, int value)
 {
 	struct sso_gpio_priv *priv = gpiochip_get_data(chip);
@@ -138,6 +147,7 @@ static int sso_gpio_gc_init(struct sso_gpio_priv *priv,
 	gc->free		= sso_gpio_free;
 	gc->get_direction	= sso_gpio_get_dir;
 	gc->direction_output	= sso_gpio_dir_out;
+	gc->get			= sso_gpio_get;
 	gc->set			= sso_gpio_set;
 
 	gc->label		= name;
